### PR TITLE
Rework experimental dependency

### PR DESCRIPTION
**Pull request by tinco**
I've replaced the experimental dependency checker with the one I built. I tested on the Beers example, for which it produces correct results now. The original experimental dependency checker did not work at all on the Beers example.

This PR is not ready for merging, just filed it so you can track my progress.

### CODE REVIEW

**Comment by tinco** [comment]
I cleaned up the code, and I added a means to effectively debug the dependency analyzers. You can now run `rake __show_dependencies` or `rake __show_dependencies experimental_dependency=true` to get constant analysis information for the project you're in. 

If we've got some more substantial projects to run that on than the Beers example we could determine if there's any important differences between the experimental and the stable dependency checker. I still feel the stable one is nicer because it achieves the same result in about half the lines, but it could be there's some weakness that inspired them to start on the experimental one.

The stable one is smaller because it doesn't have to implement code for parsing subtrees it isn't interested in, the experimental one necessarily has to implement an almost full parser.

In summary, I think it's worth to merge this branch because it's improved the existing dependency.rb with a debugging tool, and replaces the non-functional experimental_dependency.rb with a more functional one. I don't recommend switching to the experimental one, and I think it's likely we could just delete it if it turns out making dependency.rb complete enough is easy.
